src/utils/validation.ts

```diff
@@ -0,0 +1,74 @@
+import { getUserByEmail, getUserByUsername } from '../../model/user';
+
+async function usernameExists(username: string) {
+    const user = await getUserByUsername(username);
+    return user ? true : false;
+}
+
+async function emailExists(email: string) {
+    const userEmail = await getUserByEmail(email);
+    return userEmail ? true : false;
+}
+
+function validateEmail(email: string) {
+    const re = /^[^\s@]+@[^\s@]+\.[^\s@]+$/;
+    return re.test(String(email).toLowerCase());
```

Comment:
a new method for me! .test() checks the regex rule for a match

Reply:
Can't take any credit - this was all @oskarprzybylski23!